### PR TITLE
Use the new mocha hooks

### PR DESCRIPTION
**Pull request by scinos**
### Background

Mocha v8 introduced a different way to set up [Root Hooks](https://mochajs.org/#root-hook-plugins), and they recommend migrating to this new system.

We use root hooks extensively in e2e tests, most of them defined in https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/lib/mocha-hooks.js. 

### Changes proposed in this Pull Request

* Introduce Root Hooks. This PR uses hooks related to video recording as a proof of concept. If it works, I'll migrate the rest of the hooks in a separate PR.
* Refactor the Video Recorder API to make it more "hook friendly" (specifically, extract the screenshot functionality from `stopVideo()` to its own `takeScreenshot()` method.

### Testing instructions

Check that the e2e builds are still working and screenshots/videos are being captured.

### CODE REVIEW

File: test/e2e/lib/hooks/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Internal dependencies
+ */
+import { startVideo, stopVideo, takeScreenshot } from './video-recorder';
+
+export const mochaHooks = {
+	afterAll: [ stopVideo ],
```
Comment (scinos) [inline test/e2e/lib/hooks/index.js:7]: Note that `afterAll` is equivalent to the old `after` (similar for `beforeAll` and `before`)

File: test/e2e/lib/hooks/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	beforeAll: [ startVideo ],
+	afterEach: [ takeScreenshot ],
```
Comment (noahtallen) [inline test/e2e/lib/hooks/index.js:9]: The previous behavior stopped the video recorder after a failing test. Now we only stop at the end of the entire suite (?) -- does that cause a difference in functionality?

**Review by noahtallen — APPROVED** [review]
Nice, looks like screenshots and videos are still working :+1:

File: test/e2e/lib/hooks/index.js (hunk 2 above)
Comment (scinos) [reply test/e2e/lib/hooks/index.js:9]: In practice, it doesn't. We only run one suite per process, so effectively, a failing test is the end of the suite.

**Pull request merged**